resources/templates/show.twig

```diff
@@ -35,7 +56,37 @@
                                                     <a href="{{ path_for('branch', { 'virtualHost': virtualHost, 'repository': forwardSlashedRelativePath }) }}" style="color: #fff"><span aria-hidden="true" class="glyphicon glyphicon-random" style="top: 2px"></span></a>
                                                 {% endif %}
                                             </span>
-                                            <span><span class="badge">{{ repository.trackingInformation.behind }}</span></span>
+                                            <span>
+                                                {% if repository.trackingInformation.behind %}
+                                                    <span
+                                                        class="badge"
+                                                        style="cursor: pointer;"
+                                                        data-toggle="popover"
+                                                        title="{{ repository.trackingInformation.behind }} commits behind"
+                                                        data-content="
+                                                        <table>
+                                                            {% for commit in repository.trackingInformation.log.behind|slice(0, 10) %}
+                                                                <tr>
+                                                                <td><b>{{ commit.hash }}</b> &nbsp;</td>
+                                                                <td>
+                                                                {% if commit.message|length > 120 %}
+                                                                    <span title='{{ commit.message }}'>{{ commit.message[:120] }} ...</span>
+                                                                {% else %}
+                                                                     {{ commit.message }}
+                                                                {% endif %}
+
+                                                                </td>
+                                                                <tr>
+                                                            {% endfor %}
+                                                        </table>
+                                                        " >
+                                                        {{ repository.trackingInformation.behind }}
+                                                    </span>
+                                                {% else %}
+                                                    <span class="badge">{{ repository.trackingInformation.behind }}</span>
+                                                {% endif %}
+
+                                            </span>
```

Comment:
I would prevent this information from being fetch for each repository and change the request for the popover to an ajax request. So this is only loaded if needed.